Feature/add django reversion support

drnlm opened the pull request:
This uses django-reversion to track edits to talks and pages.

The history is only exposed through the admin interface. We probably want to expose it in some way for people editing pages, but I'm not sure that's required for talks. Displaying the date of last edit to people who can change talks and pages is probably a good idea.

We may want to hook this up to the Sponsors model as well. I'm not sure it makes sense for the other models we have.

hodgestar commented on wafer/talks/views.py:
```diff
@@ -104,12 +109,30 @@ def get_context_data(self, **kwargs):
+        with revisions.create_revision():
+            result = super(TalkUpdate, self).form_valid(form)
+            revisions.set_user(self.request.user)
+            revisions.set_comment("Talk Modified")
```
I'm not sure how much better this is, but I think we could do:
```python
@revisions.create_revision()
def form_valid(self, form):
    revisions.set_user(self.request.user)
    revisions.set_comment("Talk Modified")
    return super(TalkUpdate, self).form_valid(form)
```

drnlm replied:
I'm not sure how I missed the decorator option in the reversion docs.

While this is a bit more compact, I'm mainly concerned about shoving everything into form_valid. Unfortunately, I don't see a better place for this, especially as I don't want to go the ReversionMiddlewar route.

hodgestar commented:
Left a couple of points for discussion.

Agreed that we should add it for Sponsors (and Sponsor Packages) and that a separate PR is likely better for that (although I don't really mind if we do it all in one go either).

drnlm commented:
Since we only currently manage Sponsors through the admin interface, it's easy enough to hook up as well, so I've done that.

I've changed to the decorator syntax, since it is a bit more readable and there's no benefit in our use case for using the context manager syntax.

After re-reading the Django docs, I haven't found an obviously better way of hooking things up, so I've dropped that comment.

hodgestar commented:
:+1:
